tests/test_multi_file_viewer.ts

```diff
@@ -0,0 +1,999 @@
+                            </multi_file_viewer>
+                          </div>
+                        </div>`,
+            components: {
+                'multi_file_viewer': MultiFileViewer
+            }
+        })
+
+        class WrapperComponent extends Vue {
+
+            open_file(file_number: number) {
+                let mfv: MultiFileViewer = <MultiFileViewer> this.$refs.multi_file;
+                if (file_number === 1) {
+                    mfv.add_to_viewing(this.file1, this.file_1_content);
+                }
+                else if (file_number === 2) {
+                    mfv.add_to_viewing(this.file2, this.file_2_content);
+                }
+                else if (file_number === 3) {
+                    mfv.add_to_viewing(this.file3, this.file_3_content);
+                }
+                else {
+                    mfv.add_to_viewing(this.file4, this.file_4_content);
+                }
+            }
+
+            file1 = "Kiwi Header";
+
+            file2 = "Lime Header";
+
+            file3 = "Mango Header";
+
+            file4 = "Nectarine Header";
+
+            file_1_content = "Kiwi Body";
+
+            file_2_content = "Lime Body";
+
+            file_3_content = "Mango Body";
+
+            file_4_content = "Nectarine Body";
+        }
+
+        const wrapper = mount(WrapperComponent);
+        let multi_file_viewer = wrapper.find({ref: 'multi_file'});
+
+        let add_first_file_button = wrapper.find(".first-button");
+        add_first_file_button.trigger('click');
+        expect(multi_file_viewer.vm.$data.active_tab_index).toEqual(0);
+        expect(multi_file_viewer.vm.$data.files_currently_viewing.length).toEqual(1);
+        let view_file_component = multi_file_viewer.find({ref: 'view_file_component'});
+        expect(view_file_component.text()).toContain(wrapper.vm.$data.file_1_content);
+
+        let add_second_file_button = wrapper.find(".second-button");
+        add_second_file_button.trigger('click');
+        expect(multi_file_viewer.vm.$data.active_tab_index).toEqual(1);
+        expect(multi_file_viewer.vm.$data.files_currently_viewing.length).toEqual(2);
+        expect(view_file_component.text()).toContain(wrapper.vm.$data.file_2_content);
+
+        let leftmost_tab = wrapper.findAll('.tab-label').at(0);
+        leftmost_tab.trigger('click');
+        expect(multi_file_viewer.vm.$data.active_tab_index).toEqual(0);
+
+        let leftmost_tab_close_x = wrapper.findAll('.close-x');
+        leftmost_tab_close_x.trigger('click');
+
+        multi_file_viewer = wrapper.find({ref: 'multi_file'});
+
+        expect(multi_file_viewer.vm.$data.active_tab_index).toEqual(0);
+        expect(multi_file_viewer.vm.$data.files_currently_viewing.length).toEqual(1);
+        expect(view_file_component.text()).toContain(wrapper.vm.$data.file_2_content);
+
+    });
+
+    test('When the only tab gets deleted, the active_tab_index becomes negative', () => {
```

Comment:
I think the tabs component has a special check for when there are no tabs present. Inside the tabs component, the selected tab index is 0 when that's the case. Did something go wrong when there were no tabs open but the active tab index was 0?

Reply:
For some reason when testing the multi_file_viewer (and the active_tab_index is initially set to 0) the tab body doesn't show up until you've added a second file. That's the only reason I kept around the -1. I'm not really sure why that happens.

Reply:
In just the test?

Reply:
Just pushed a fix for this to this branch